classes/galley/DAO.inc.php

```diff
@@ -0,0 +1,308 @@
+        'isApproved' => 'is_approved',
+        'locale' => 'locale',
+        'label' => 'label',
+        'publicationId' => 'publication_id',
+        'seq' => 'seq',
+        'urlPath' => 'url_path',
+        'urlRemote' => 'remote_url',
+        'doiId' => 'doi_id',
+    ];
+
+    public function newDataObject(): Galley
+    {
+        return app(Galley::class);
+    }
+
+    public function get(int $id): ?Galley
+    {
+        return parent::get($id);
+    }
+
+    public function getByUrlPath(string $urlPath, Publication $publication): ?Galley
+    {
+        $row = DB::table($this->table)
+            ->where('publication_id', $publication->getId())
+            ->where('url_path', $urlPath)
+            ->first();
+        return $row ? $this->fromRow($row) : null;
+    }
+
+    /**
+     * Get the number of galleys matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('g.' . $this->primaryKeyColumn)
+            ->get()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('g.' . $this->primaryKeyColumn)
+            ->pluck('g.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of galleys matching the configured query
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->select(['g.*'])
+            ->get();
+
+        return LazyCollection::make(function () use ($rows) {
+            foreach ($rows as $row) {
+                yield $this->fromRow($row);
+            }
+        });
+    }
+
+    public function fromRow(object $row): Galley
+    {
+        $galley = parent::fromRow($row);
+
+        if (!empty($galley->getData('doiId'))) {
+            $galley->setData('doiObject', Repo::doi()->get($galley->getData('doiId')));
+        }
+
+        return $galley;
+    }
+
+    public function insert(Galley $galley): int
+    {
+        return parent::_insert($galley);
+    }
+
+    public function update(Galley $galley)
+    {
+        parent::_update($galley);
+    }
+
+    public function delete(Galley $galley)
+    {
+        parent::_delete($galley);
+    }
+
+    /** @copydoc RepresentationDAOInterface::getById() */
+    public function getById(int $galleyId, ?int $publicationId = null, ?int $contextId = null): ?Galley
+    {
+        return $this->get($galleyId);
+    }
+
+    /** @copydoc RepresentationDAOInterface::getByPublicationId() */
+    public function getByPublicationId(int $publicationId): array
+    {
+        return Repo::galley()->getMany(
+            Repo::galley()
+                ->getCollector()
+                ->filterByPublicationIds([$publicationId])
+        )->toArray();
```

Comment:
Are these functions for the sake of keeping `RepresentationDAOInterface` compatible with the previous `RepresentationDAO`? I don't mind if we break this, given that we're already changing so much in 3.4, in order to encourage standardization on the new EntityDAO toolset, LazyCollections, etc.

Reply:
Things got really messy with `PublicationFormatDAO` in OMP. There are several places that do things like:

```php
$representations = Application::getRepresentationDAO()->callSomeFunction(...);
```

The `RepresentationDAOInterface` documents the functions that get called against `GalleyDAO` and `PublicationFormatDAO`, from pkp-lib when we don't know which one is being used. I hope that #6702 will allow us to remove the `Application::getRepresentationDAO()` stuff, and therefore this interface.